src/overlays/actors/ovl_En_Baisen/z_en_baisen.c

```diff
@@ -21,30 +38,255 @@ const ActorInit En_Baisen_InitVars = {
+        if (this->animIndex == 0) {
+            EnBaisen_ChangeAnimation(this, 1);
+        }
+    } else {
+        this->unk29E = Math_Vec3f_Yaw(&this->actor.world.pos, &this->unk2A4->world.pos);
+        if (this->animIndex != 0) {
+            EnBaisen_ChangeAnimation(this, 0);
+        }
+    }
+    if ((globalCtx->msgCtx.unk11F04 == 0x2AC6) || (globalCtx->msgCtx.unk11F04 == 0x2AC7) ||
+        (globalCtx->msgCtx.unk11F04 == 0x2AC8)) {
+        this->skelAnime.animPlaybackSpeed = 0.0f;
+        this->unk29E = this->actor.yawTowardsPlayer;
+    }
+    if (this->unk2AC == 2) { // Note: This variable is only ever set to 1.
+        func_80BE87FC(this);
+    }
+}
 
-#pragma GLOBAL_ASM("./asm/non_matchings/overlays/ovl_En_Baisen_0x80BE84F0/func_80BE87FC.asm")
+void func_80BE8AAC(EnBaisen* this, GlobalContext* globalCtx) {
+    if (this->textIdIndex & 1) {
+        this->unk29E = this->actor.world.rot.y;
+        if (this->animIndex == 0) {
+            EnBaisen_ChangeAnimation(this, 1);
+        }
+    } else {
+        if (this->unk2A4 != NULL) {
+            this->unk29E = Math_Vec3f_Yaw(&this->actor.world.pos, &this->unk2A4->world.pos);
+        }
+        if (this->animIndex != 0) {
+            EnBaisen_ChangeAnimation(this, 0);
+        }
+    }
+    if ((func_80152498(&globalCtx->msgCtx) == 5) && func_80147624(globalCtx)) {
+        func_801477B4(globalCtx);
+        this->textIdIndex++;
+        if (this->textIdIndex < 6) {
+            func_80151938(globalCtx, textIds[this->textIdIndex]);
+            if (!(this->textIdIndex & 1)) {
+                this->unk2A4 = this->heishiPointer;
+            } else {
+                this->unk2A4 = &this->actor;
+            }
+            func_800B86C8(this->unk2A4, globalCtx, this->unk2A4);
+        } else {
+            func_80BE87FC(this);
+        }
+    }
+}
 
-#pragma GLOBAL_ASM("./asm/non_matchings/overlays/ovl_En_Baisen_0x80BE84F0/func_80BE887C.asm")
+void EnBaisen_Update(Actor* thisx, GlobalContext* globalCtx) {
+    s32 pad;
+    EnBaisen* this = THIS;
 
-#pragma GLOBAL_ASM("./asm/non_matchings/overlays/ovl_En_Baisen_0x80BE84F0/func_80BE895C.asm")
+    SkelAnime_FrameUpdateMatrix(&this->skelAnime);
+    if (this->unusedCounter != 0) {
+        this->unusedCounter--;
+    }
+    this->actor.shape.rot.y = this->actor.world.rot.y;
+    if ((this->paramCopy != 0) && (gSaveContext.day == 3) && (gSaveContext.isNight)) {
+        Actor_MarkForDeath(&this->actor);
+        return;
+    }
+    this->actionFunc(this, globalCtx);
+    Actor_SetVelocityAndMoveYRotationAndGravity(&this->actor);
+    Actor_UpdateBgCheckInfo(globalCtx, &this->actor, 20.0f, 20.0f, 50.0f, 0x1D);
+    Actor_SetScale(&this->actor, 0.01f);
+    if (this->unk290 != 0) {
+        func_80BE871C(this);
+    }
+    Actor_SetHeight(&this->actor, 60.0f);
+    Math_SmoothStepToS(&this->headRotX, this->headRotXTarget, 1, 3000, 0);
+    Math_SmoothStepToS(&this->headRotY, this->headRotYTarget, 1, 1000, 0);
```

Comment:
```suggestion
    Math_SmoothStepToS(&this->headRotX, this->headRotXTarget, 1, 0xBB8, 0);
    Math_SmoothStepToS(&this->headRotY, this->headRotYTarget, 1, 0x3E8, 0);
```
I know in OoT it's okay to move angles to decimal if the numbers are round, but I think in MM we've been trying to keep all angles in hex?